homeassistant/components/mqtt/__init__.py

```diff
@@ -300,6 +305,67 @@ def _convert_outgoing_payload(
         )
 
 
+class MqttValueTemplate:
+    """Class for rendering MQTT value template with possible json values."""
+
+    def __init__(
+        self,
+        value_template: template.Template | None,
+        *,
+        hass: HomeAssistant | None = None,
+        entity: Entity | None = None,
+        config_attributes: template.TemplateVarsType = None,
+    ) -> None:
+        """Instantiate a value template."""
+        self._attr_value_template = value_template
+        self._config_attributes = config_attributes
+        if value_template is None:
+            return
+
+        value_template.hass = hass
+        self._entity = entity
+
+        if entity:
+            value_template.hass = entity.hass
+
+    @callback
+    def async_render_with_possible_json_value(
+        self,
+        payload: ReceivePayloadType,
+        default: ReceivePayloadType | object = _SENTINEL,
+        variables: template.TemplateVarsType = None,
+    ) -> ReceivePayloadType:
+        """Render with possible json value or pass-though a received MQTT value."""
+        if self._attr_value_template is None:
+            return payload
+
+        values: dict[str, Any] = {}
+
+        if variables is not None:
+            values.update(variables)
+
+        if self._config_attributes is not None:
+            values.update(self._config_attributes)
+
+        if self._entity:
+            values[ATTR_ENTITY_ID] = self._entity.entity_id
+            values[ATTR_NAME] = self._entity.name
+
+        if default == _SENTINEL:
+            return self._attr_value_template.async_render_with_possible_json_value(
+                payload, variables=values or None
+            )
+
+        return self._attr_value_template.async_render_with_possible_json_value(
+            payload, default, variables=values
+        )
+
+    @property
+    def value_template(self) -> template.Template | None:
+        """Return the active value template or None."""
+        return self._attr_value_template
```

Comment:
How come this is not covered by tests? Is the `value_template` method not used?

Reply:
property is not in use, removed it.